packages/dart/noports_core/lib/src/srvd/isolates/relay_worker.dart

```diff
@@ -0,0 +1,181 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:isolate';
+
+import 'package:at_utils/at_logger.dart';
+import 'package:meta/meta.dart';
+import 'package:noports_core/src/srvd/isolates/types.dart';
+import 'package:noports_core/sshnp_foundation.dart';
+import 'package:noports_core/src/srvd/relay_auth_verifiers.dart';
+import 'package:noports_core/src/srvd/srvd_session_params.dart';
+
+typedef RelayWorkerRequestHandler = Future<dynamic> Function(IIRequest);
+
+abstract class RelayWorker implements RelayAuthVerifyHelper {
+  final SendPort toMain;
+  final bool logTraffic;
+  final bool verbose;
+  final String loggingTag;
+  late final ReceivePort fromMain;
+  late final AtSignLogger logger;
+  final Map<String, RelayWorkerRequestHandler> reqHandlers = {};
+  final Map<int, Completer<IIResponse>> rpcCompleters = {};
+
+  RelayWorker({
+    required this.toMain,
+    required this.logTraffic,
+    required this.verbose,
+    required this.loggingTag,
+  }) {
+    AtSignLogger.defaultLoggingHandler = AtSignLogger.stdErrLoggingHandler;
+    AtSignLogger.root_level = verbose ? 'INFO' : 'WARNING';
+    logger = AtSignLogger(' srvd / $runtimeType / $loggingTag ');
+
+    // Make a ReceivePort so the main isolate can send messages to us
+    // and send it to the main isolate
+    fromMain = ReceivePort(loggingTag);
+    toMain.send(fromMain.sendPort);
+
+    startListening();
+  }
+
+  Future<void> stop([IIRequest? req]);
+
+  Future<void> run();
+
+  @nonVirtual
+  void startListening() {
```

Comment:
subclasses are expected to add to `reqHandlers` to register functions so that this function can dispatch IIRequest messages correctly